bindings/generate_bindings.py

```diff
@@ -0,0 +1,684 @@
+
+    # only $refs don't have types
+    assert "type" in schema, (path, schema)
+
+    if schema["type"] == "string":
+        return String()
+
+    if schema["type"] == "integer":
+        return Int()
+
+    if schema["type"] == "boolean":
+        return Bool()
+
+    if schema["type"] in ("float", "number"):
+        return Float()
+
+    if schema["type"] == "object" and "properties" not in schema:
+        adlProps = schema.get("additionalProperties")
+        if adlProps is None:
+            return Dict(Any())
+        return Dict(classify_type(enums, path + ".additionalProperties", adlProps))
+
+    if schema["type"] == "array":
+        items = schema.get("items")
+        if items is None:
+            raise ValueError(path, schema)
+            return Sequence(Any())
+        return Sequence(classify_type(enums, path + ".items", items))
+
+    raise ValueError(f"unhandled schema: {schema} @ {path}")
+
+
+def process_enums(swagger_definitions: dict) -> typing.Dict[int, str]:
+    """
+    Process enums from swagger definitions. In OpenAPI spec v2 generated
+    by  protoc-gen-openapi enums are not linked to a definition and are inlined.
+    Here we preprocess them so that they can  be linked to a definition.
+    """
+    enums = {}
+    for name, schema in swagger_definitions.items():
+        if "enum" in schema:
+            members = schema["enum"]
+            enums[json.dumps(members)] = name
+    return enums
+
+
+def process_definitions(swagger_definitions: dict, enums: dict) -> TypeDefs:
+    defs = {}  # type: TypeDefs
+    for name, schema in swagger_definitions.items():
+        path = name
+        if "enum" in schema:
+            if schema["type"] == "string":
+                members = schema["enum"]
+                defs[name] = Enum(name, members)
+                continue
+            raise ValueError("unhandled enum type ({schema['type']}): {schema}")
+
+        if schema["type"] == "object":
+            # top-level named objects should be classes, not typed dictionaries:
+            assert "additionalProperties" not in schema, (name, schema)
+            if "properties" in schema:
+                required = set(schema.get("required", []))
+                members = {
+                    k: Parameter(
+                        k, classify_type(enums, path, v), (k in required), "definitions"
+                    )
+                    for k, v in schema["properties"].items()
+                }
+                defs[name] = Class(name, members)
+                continue
+            else:
+                # empty responses or empty requests... we don't care.
+                defs[name] = None
+                continue
+        raise ValueError(f"unhandled schema: {schema} @ {path}")
+    return defs
+
+
+def is_expected_path(text: str) -> bool:
+    """
+    check if there are any dots appearing outside of curly braces if any.
+    assuming there are no nested curly braces.
```

Comment:
nit: the docstring is good but let's make it proper grammar (complete sentences and capitalizations).